src/Core/Configurations/HostedRuntimeConfigProvider.cs

```diff
@@ -10,45 +10,34 @@
 using Azure.DataApiBuilder.Config.NamingPolicies;
 using Azure.DataApiBuilder.Config.ObjectModel;
 using Azure.DataApiBuilder.Service.Exceptions;
+using static Azure.DataApiBuilder.Core.Configurations.IRuntimeConfigProvider;
 
 namespace Azure.DataApiBuilder.Core.Configurations;
 
 /// <summary>
-/// This class is responsible for exposing the runtime config to the rest of the service.
-/// The <c>RuntimeConfigProvider</c> won't directly load the config, but will instead rely on the <see cref="FileSystemRuntimeConfigLoader"/> to do so.
+/// This class is responsible for exposing the runtime config to the rest of the service when in a hosted scenario.
+/// The <c>HostedRuntimeConfigProvider</c> won't directly load the config, but will instead rely on the <see cref="FileSystemRuntimeConfigLoader"/> to do so.
 /// </summary>
 /// <remarks>
-/// The <c>RuntimeConfigProvider</c> will maintain internal state of the config, and will only load it once.
+/// The <c>HostedRuntimeConfigProvider</c> will maintain internal state of the config, and will only load it once.
 ///
 /// This class should be treated as the owner of the config that is available within the service, and other classes
 /// should not load the config directly, or maintain a reference to it, so that we can do hot-reloading by replacing
 /// the config that is available from this type.
 /// </remarks>
-public class RuntimeConfigProvider
+public class HostedRuntimeConfigProvider : IRuntimeConfigProvider
 {
-    public delegate Task<bool> RuntimeConfigLoadedHandler(RuntimeConfigProvider sender, RuntimeConfig config);
-
     public List<RuntimeConfigLoadedHandler> RuntimeConfigLoadedHandlers { get; } = new List<RuntimeConfigLoadedHandler>();
 
-    /// <summary>
-    /// Indicates whether the config was loaded after the runtime was initialized.
-    /// </summary>
-    /// <remarks>This is most commonly used when DAB's config is provided via the <c>ConfigurationController</c>, such as when it's a hosted service.</remarks>
     public bool IsLateConfigured { get; set; }
 
-    /// <summary>
-    /// The access tokens representing a Managed Identity to connect to the database.
-    /// The key is the unique datasource name and the value is the access token.
-    /// </summary>
-    public Dictionary<string, string?> ManagedIdentityAccessToken { get; private set; } = new Dictionary<string, string?>();
-
-    public RuntimeConfigLoader ConfigLoader { get; private set; }
+    public Dictionary<string, string?> ManagedIdentityAccessToken { get; set; } = new Dictionary<string, string?>();
 
-    private ConfigFileWatcher? _configFileWatcher;
+    public RuntimeConfigLoader ConfigLoader { get; set; }
 
-    private RuntimeConfig? _runtimeConfig;
+    public RuntimeConfig? _runtimeConfig;
 
-    public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
+    public HostedRuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     {
         ConfigLoader = runtimeConfigLoader;
```

Comment:
It is used for testing because our tests are using the initialize function, which is for the hosted scenario. I am looking into how to refactor the tests to eliminate this, but the answer is not obvious.